DatadogSessionReplay/Sources/SessionReplay.swift

```diff
@@ -50,8 +80,41 @@ public enum SessionReplay {
 
         core.telemetry.configuration(
             defaultPrivacyLevel: configuration.defaultPrivacyLevel.rawValue,
-            sessionReplaySampleRate: Int64(withNoOverflow: configuration.replaySampleRate)
+            sessionReplaySampleRate: Int64(withNoOverflow: configuration.replaySampleRate),
+            startRecordingImmediately: configuration.startRecordingImmediately
         )
     }
+
+    internal static func startRecording(core: DatadogCoreProtocol) throws {
+        guard !(core is NOPDatadogCore) else {
+            throw ProgrammerError(
+                description: "Datadog SDK must be initialized before calling `SessionReplay.startRecording()`."
+            )
+        }
+
+        guard let sr = core.get(feature: SessionReplayFeature.self) else {
+            throw ProgrammerError(
+                description: "Session Replay must be initialized before calling `SessionReplay.startRecording()`."
+            )
+        }
+
+        sr.startRecording()
+    }
+
+    internal static func stopRecording(core: DatadogCoreProtocol) throws {
+        guard !(core is NOPDatadogCore) else {
+            throw ProgrammerError(
+                description: "Datadog SDK must be initialized before calling `SessionReplay.stopRecording()`."
+            )
+        }
+
+        guard let sr = core.get(feature: SessionReplayFeature.self) else {
+            throw ProgrammerError(
+                description: "Session Replay must be initialized before calling `SessionReplay.stopRecording()`."
+            )
+        }
+
+        sr.stopRecording()
+    }
```

Comment:
/question do we need this level of logs? When starting we could only check if the feature is enabled, not necessarily if core is initialised. And stopping a no-op feature is not necessarily a programmer error, it could be silent.

Reply:
Good point, I'll make the change.